server/src/main/java/org/elasticsearch/health/stats/HealthApiStats.java

```diff
@@ -0,0 +1,144 @@
+ * diagnoses. An example of what the stats look like is seen below. The stats are exposed via the xpack usage api for telemetry.
+ * {
+ *   "invocations": {
+ *     "total": 22,
+ *     "verbose_true": 12,
+ *     "verbose_false": 10
+ *   },
+ *   "statuses": {
+ *     "green": 10,
+ *     "yellow": 4,
+ *     "red": 8,
+ *     "values": ["green", "yellow", "red"]
+ *   },
+ *   "indicators": {
+ *     "red" : {
+ *       "master_stability": 2,
+ *       "ilm":2,
+ *       "slm": 4,
+ *       "values": ["master_stability", "ilm", "slm"]
+ *     },
+ *     "yellow": {
+ *       "disk": 1,
+ *       "shards_availability": 1,
+ *       "master_stability": 2,
+ *       "values": ["disk", "shards_availability", "master_stability"]
+ *     }
+ *   },
+ *   "diagnoses": {
+ *     "red": {
+ *       "elasticsearch:health:shards_availability:primary_unassigned": 1,
+ *       "elasticsearch:health:disk:add_disk_capacity_master_nodes": 3,
+ *       "values": [
+ *         "elasticsearch:health:shards_availability:primary_unassigned",
+ *         "elasticsearch:health:disk:add_disk_capacity_master_nodes"
+ *       ]
+ *     },
+ *     "yellow": {
+ *       "elasticsearch:health:disk:add_disk_capacity_data_nodes": 1,
+ *       "values": ["elasticsearch:health:disk:add_disk_capacity_data_nodes"]
+ *     }
+ *   }
+ * }
+ */
+public class HealthApiStats {
+
+    private static final String TOTAL_INVOCATIONS = "invocations.total";
+    private static final String VERBOSE_TRUE = "invocations.verbose_true";
+    private static final String VERBOSE_FALSE = "invocations.verbose_false";
+    private final Function<HealthStatus, String> statusLabel = status -> Strings.format("statuses.%s", status.xContentValue());
+    private final BiFunction<HealthStatus, String, String> indicatorLabel = (status, indicator) -> Strings.format(
+        "indicators.%s.%s",
+        status.xContentValue(),
+        indicator
+    );
+    private final BiFunction<HealthStatus, String, String> diagnosisLabel = (status, diagnosis) -> Strings.format(
+        "diagnoses.%s.%s",
+        status.xContentValue(),
+        diagnosis
+    );
+
+    private final Set<HealthStatus> statuses = ConcurrentHashMap.newKeySet();
+    private final Map<HealthStatus, Set<String>> indicators = new ConcurrentHashMap<>();
+    private final Map<HealthStatus, Set<String>> diagnoses = new ConcurrentHashMap<>();
+    private final Counters counters = new Counters(TOTAL_INVOCATIONS);
+
+    public HealthApiStats() {}
+
+    public void track(boolean verbose, GetHealthAction.Response response) {
+        counters.inc(TOTAL_INVOCATIONS);
+        if (verbose) {
+            counters.inc(VERBOSE_TRUE);
+        } else {
+            counters.inc(VERBOSE_FALSE);
+        }
+        HealthStatus status = response.getStatus() != null
+            ? response.getStatus()
+            : response.getIndicators().stream().map(HealthIndicatorResult::status).findFirst().orElse(null);
```

Comment:
shall we document this status check is due to an API drill down call? (the why in doing this, especially the `findFirst` - and only - call, is not immediately apparent)

Reply:
yes